lib/eslint.js

```diff
@@ -998,6 +998,11 @@ module.exports = (function() {
             i,
             len;
 
+        // Special Node.js scope means we need to start one level deeper
+        if (scope.type === "global" && currentConfig.ecmaFeatures.globalReturn) {
```

Comment:
We also need to check this scope is the ecmaFeature `modules` is used.

I think the condition should be:

```javascript
 if (scope.type === "global" (currentConfig.ecmaFeatures.globalReturn || currentConfig.ecmaFeatures.modules)) {
```